termd-build-driver/src/main/java/org/jboss/pnc/termdbuilddriver/TermdBuildDriver.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.termdbuilddriver;
+
+import org.jboss.logging.Logger;
+import org.jboss.pnc.model.BuildConfiguration;
+import org.jboss.pnc.model.BuildType;
+import org.jboss.pnc.spi.builddriver.BuildDriver;
+import org.jboss.pnc.spi.builddriver.BuildDriverStatus;
+import org.jboss.pnc.spi.builddriver.RunningBuild;
+import org.jboss.pnc.spi.builddriver.exception.BuildDriverException;
+import org.jboss.pnc.spi.environment.RunningEnvironment;
+import org.jboss.pnc.termdbuilddriver.commands.TermdCommandInvoker;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+@ApplicationScoped
+public class TermdBuildDriver implements BuildDriver {
+
+    public static final String DRIVER_ID = "TermdBuildDriver";
+
+    private static final Logger logger = Logger.getLogger(TermdBuildDriver.class);
+
+    public TermdBuildDriver() {
+    }
+
+    @Override
+    public String getDriverId() {
+        return DRIVER_ID;
+    }
+
+    @Override
+    public boolean canBuild(BuildType buildType) {
+        return BuildType.JAVA.equals(buildType);
+    }
+
+    @Override
+    public RunningBuild startProjectBuild(BuildConfiguration buildConfiguration, final RunningEnvironment runningEnvironment)
+            throws BuildDriverException {
+
+        TermdRunningBuild termdRunningBuild = new TermdRunningBuild(runningEnvironment, buildConfiguration);
+        TermdCommandInvoker invoker = new TermdCommandInvoker(URI.create(runningEnvironment.getJenkinsUrl()));
+
+        checkoutSources(termdRunningBuild, invoker)
+                .thenCompose(returnedTermdRunningBuild -> invokeBuild(returnedTermdRunningBuild, invoker))
+                .thenCompose(returnedTermdRunningBuild -> completeBuild(returnedTermdRunningBuild, invoker));
+
```

Comment:
You should probably catch exceptions from performCommand and mark build as failed when exceptions occur ?